### DISCUSSION TITLE
balance/balance4

### DISCUSSION THREAD

**Discussion by giscus** [isno/theByteBook#243] [General]
# balance/balance4

构建大规模高可用的分布式系统

https://www.thebyte.com.cn/balance/balance4.html

**Comment by 8Haixin** [comment]
“4.5.5 一致性哈希容错模式”，看的好懵。

**Reply by isno** [reply]
懵是对的，这块我没写好。

**Reply by isno** [reply]
这块重写了

**Comment by lavender1900** [comment]
老师，我看了这篇文章后进行了一些搜索，基本上搜索到的文章都说VIP是内部网络使用的，不属于public ip，但是显然老师您的图中，负载均衡器是直接对外服务的，所以负载均衡器的VIP肯定是公网IP，这个又跟我搜索的文章内容矛盾。所以负载均衡器到底是怎么实现VIP的呢？

**Reply by isno** [reply]
负载均衡器对外，VIP 就是个 public ip。对外，那就是个内网 IP。

VIP 的实现几句话说不清楚，跟 ARP、lo 接口有关系，你搜下网上的内容。
